gensim/models/phrases.py

```diff
@@ -671,10 +671,15 @@ def npmi_scorer(worda_count, wordb_count, bigram_count, len_vocab, min_count, co
     where :math:`prob(word) = \\frac{word\_count}{corpus\_word\_count}`
 
     """
-    pa = worda_count / corpus_word_count
-    pb = wordb_count / corpus_word_count
-    pab = bigram_count / corpus_word_count
-    return log(pab / (pa * pb)) / -log(pab)
+    if bigram_count > min_count:
+        pa = worda_count / corpus_word_count
+        pb = wordb_count / corpus_word_count
+        pab = bigram_count / corpus_word_count
+        return log(pab / (pa * pb)) / -log(pab)
+    else:
+        # Return the value below minimal npmi, to make sure that phrases
+        # will be created only out of bigrams more frequent than min_count
+        return -1.1
```

Comment:
What is the reasoning behind -1.1? Looks too magical.

Deserves a comment at the very least, but maybe a principled value would be even better? `-inf`? `None`?

Reply:
That value will be compared against the threshold parameter that was supplied by the user, which even though is not restricted, should fall within -1 and 1.

Reply:
Good catch.

I'd be in favor of returning ```None``` (for all scoring functions) rather than a number for any ```min_count``` violation since that would mean all scorers return the same value that means "min_count not met". But I also don't know if ```None``` may break things. What tests fail with ```None```? From a quick look this is probably safe. 

I don't think returning -1.1 is a great idea either, but I agree about it being commented at the very least.

Reply:
It seems you can't compare int/float with NoneType (at least in python 3.6.2), so [line 168](https://github.com/RaRe-Technologies/gensim/blob/cc441b7b8bf2f80b17cf44acbcb425d529876ff1/gensim/models/phrases.py#L168) of phrases.py would have to be something like `if score is not None and score > threshold:`.

Other solution, instead of returning `None` in scorers, could be `return -float('Inf')` which would result in 'False' when comparing `score > threshold`.

I personally like solution 1 more.

Reply:
I am sorry for a slow reply. The pull-request woke up in a pretty busy time for me.

Thank you for remarks. Indeed -inf sounds a bit less fragile-magical. Code updated.
 
Personally, I would avoid returning None from function expected to return a number and cluttering the rest of the code.

Reply:
+1 to -Inf, I learned something :). LGTM.